psamm/datasource/native.py

```diff
@@ -174,28 +174,39 @@ class NativeModel(object):
     directory subtree that specifies part of the model.
     """
 
-    def __init__(self, path):
-        """Create a model from the specified model file or directory"""
-
-        if os.path.isfile(path):
-            self._context = FilePathContext(path)
-            with open(self._context.filepath, 'r') as f:
-                self._model = yaml_load(f)
-        else:
-            # Try to open the default file
-            for filename in DEFAULT_MODEL:
-                try:
-                    self._context = FilePathContext(
-                        os.path.join(path, filename))
-                    with open(self._context.filepath, 'r') as f:
-                        self._model = yaml_load(f)
-                        break
-                except:
-                    logger.debug('Failed to load model file', exc_info=True)
+    def __init__(self, model_from):
+        """Create a model from the specified model file, directory,
+            long string, open file, or dictionary"""
```

Comment:
Docstrings should be formatter according to https://www.python.org/dev/peps/pep-0257/ (it's similar to git commit messages). First line should be a summary (no line breaks!), then a blank line, then a longer description if necessary.